TestVectorsAwsCryptographicMaterialProviders/runtimes/python/fuzz_generator.py

```diff
@@ -0,0 +1,418 @@
+    """
+    Generate multiple fuzzed test vectors and collect new keys.
+    
+    Args:
+        num_vectors: Number of test vectors to generate
+        
+    Returns:
+        Tuple containing:
+        - Dictionary containing the test vectors
+        - Dictionary containing new keys to add to keys.json
+    """
+    test_vectors = {}
+    new_keys = {}
+    
+    print(f"Generating {num_vectors} fuzzed test vectors...")
+    print("Focus: encryptionContext, requiredEncryptionContextKeys, reproducedEncryptionContext, and key identifiers")
+    
+    for i in range(num_vectors):
+        # Use Hypothesis to generate a test vector
+        test_vector = fuzz_test_vector().example()
+        test_id = str(uuid.uuid4())
+        test_vectors[test_id] = test_vector
+        
+        # Extract new keys from raw keyring test vectors
+        if (test_vector.get("encryptKeyDescription", {}).get("type") == "raw"):
+            key_name = test_vector["encryptKeyDescription"]["key"]
+            
+            # Generate the key material for this fuzzed key name
+            # Determine the base key type from the key name pattern
+            if "aes-128" in key_name:
+                base_key_id = "aes-128"
+            elif "aes-192" in key_name:
+                base_key_id = "aes-192"
+            elif "aes-256" in key_name:
+                base_key_id = "aes-256"
+            else:
+                base_key_id = "aes-256"  # Default fallback
+            
+            
+            key_bits_map = {
+                "aes-128": 128,
+                "aes-192": 192,
+                "aes-256": 256
+            }
+            
+            base_materials = {
+                "aes-128": "AAECAwQFBgcICRAREhMUFQ==",  # 16 bytes for AES-128
+                "aes-192": "AAECAwQFBgcICRAREhMUFRYXGBkgISIj",  # 24 bytes for AES-192
+                "aes-256": "AAECAwQFBgcICRAREhMUFRYXGBkgISIjJCUmJygpMDE="  # 32 bytes for AES-256
+            }
+            
+            new_keys[key_name] = {
+                "encrypt": True,
+                "decrypt": True,
+                "algorithm": "aes",
+                "type": "symmetric",
+                "bits": key_bits_map.get(base_key_id, 256),
+                "encoding": "base64",
+                "material": base_materials.get(base_key_id, base_materials["aes-256"]),
+                "key-id": base_key_id
+            }
+        
+        if (i + 1) % 10 == 0:
+            print(f"Generated {i + 1}/{num_vectors} vectors...")
+    
+    print(f"Generated {len(new_keys)} new fuzzed keys for keys.json")
+    return test_vectors, new_keys
+
+
+def main():
+    """Main function to generate fuzzed test vectors."""
+    print("=== AWS Cryptographic Material Providers Fuzz Testing Generator ===")
+    print("Strategy: Focus on encryption context UTF-8 vulnerabilities and key identifier fuzzing")
+    
+    # Generate test vectors and new keys
+    test_vectors, new_keys = generate_fuzz_test_vectors(num_vectors=5)
```

Comment:
Let's put a TODO saying we have to increase number of vectors. 

I expectation is we generate millions of vectors and run it on CI that way we can get profit out of it. But, while developing you of course want to generate only few because it gets easier for you.